lib/palindrome_check.rb

```diff
@@ -1,7 +1,24 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear, O(n) - where n is the length of the input string
+#                  divided by 2
+# Space complexity: constant, O(1) - the space used remains the same regardless
+#                   of the string input size
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  return false if !my_phrase
+
+  counter = 0
+  start_index = 0
+  end_index = my_phrase.length - 1
```

Comment:
Liking the self explanatory variable names! :smile:

These are helpful in interviews, too - it's easier to keep track (for both you and the interviewer) of what your code's doing